lib/calculator.rb

```diff
@@ -1,52 +1,4 @@
 class Calculator 
-    def initialize         
-        a = get_number("Enter the first number: ")
-        method = get_method
-        b = get_number("Enter the second number: ")
-
-        result = calculate(a, b, method)
-
-        puts "\nResult: #{a} #{method} #{b} = #{result}"
-    end
-
-    def numeric(number)
-        !!Float(number) rescue false
-    end
-
-    def get_number(message)
-        number_is_valid = false
-
-        while !number_is_valid
-            puts "\n#{message}"
-            number = gets.chomp
-
-            if numeric(number)
-                number_is_valid = true
-            else
-                puts "Invalid number"
-            end
-        end
-
-        number.to_f
-    end
-
-    def get_method
-        method_is_valid = false
-
-        while !method_is_valid
-            puts "\nPlease enter a valid operation symbol: \nEnter + for Sum\nEnter - for Substraction\nEnter / for Division\nEnter * for Multiplication"
-            method = gets.chomp
-            
-            if method === '+' || method === '-' || method === '/' || method === '*'
-                method_is_valid = true
-            else
-                puts "\nOperation symbol is invalid"
-            end
-        end
-
-        method
-    end
-
     def calculate(a, b, method)
         case method
         when "+"
```

Comment:
A nossa calculadora atualmente suporta apenas as quatro operações básicas conforme os requisitos, mas pode ser que os requisitos mudem e seja preciso adicionar novas operações, por exemplo, calcular porcentagem ou raiz quadrada. Pra adicionar novas operações seria necessário adicionar uma condicional nesse método `calculate` e um novo método no `Calculator`.

Pra deixar o design dessa calculadora mais flexível permitindo adicionar novas operações com maior facilidade, o que acha de adicionarmos uma classe `Calculator::Operation` pra representar uma operação da calculadora?

Pensei na seguinte interface:
```suggestion
    def calculate(method, *numbers)
      Calculator::Operation.instance_for(method).call(*numbers)
    end    
```

 O que acha?

Reply:
A ideia aqui o `Calculator::Operation.instance_for` ser um Factory Method (https://refactoring.guru/design-patterns/factory-method) que vai retornar uma operação específica (por exemplo: `Calculator::Operation::Sum`) baseado no valor do `method`.